### PR TITLE
NVMe Node Changes

### PR DESCRIPTION
**Pull request by harshitap26**
# Description
Changes made on the NVMe host creation and nodeGetInfo changes
The PR is dependent on the below points - 
1. The checks will fail as local references have been used for gonvme and gobrick
2. The gonvme local reference will be remove once the library is made public
3. The gobrick reference will be removed once the gobrick changes go in - https://github.com/dell/gobrick/pull/8

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/158|

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested on linux environment with nvme library installed
- [x] Tested the gonvme library functions.
- [x] Tested the NVMe host creation and NodeGetInfo changes.

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -111,24 +115,38 @@ func (s *Service) Init() error {
...
+			s.useISCSI = true
 			s.useFC = false
 		case common.FcTransport:
 			if len(fcInitiators) == 0 {
 				return fmt.Errorf("FC transport was requested but FC initiator is not available")
 			}
 			s.useFC = true
+			s.useISCSI = false
 		default:
+			s.useISCSI = len(iscsiInitiators) > 0
```
Comment (spriya-m) [inline pkg/node/node.go:137]: incase both iscsi and fc initiators are present, useISCSI and useFC will be set to true here. Is it expected?

**Review by spriya-m — DISMISSED** [review]

**Review by harishp8889 — APPROVED** [review]

**Review by francis-nijay — APPROVED** [review]

**Review by harishp8889 — APPROVED** [review]

**Comment by prablr79** [comment]
@harishp8889 can you resolve the conflict ?

**Comment by harshitap26** [comment]
> @harishp8889 can you resolve the conflict ?

Removed the conflicts

**Review by spriya-m — APPROVED** [review]

**Comment by prablr79** [comment]
@francis-nijay please merge the gobrick and then you can merge this PR lets makes sure we have clean go sec and go vetting for this PR before merge

**Pull request merged**